apps/condo/domains/property/components/panels/Builder/MapConstructor.ts

```diff
@@ -0,0 +1,490 @@
+            this.editMode = 'editSection'
+        }        
+    }
+    public getSelectedSection (): BuildingSection {
+        return this.selectedSection
+    }
+    public isSectionSelected (id: string): boolean {
+        return this.selectedSection && this.selectedSection.id === id
+    }
+
+    private selectedUnit: BuildingUnit    
+    public setSelectedUnit (unit: BuildingUnit): void {
+        if (this.isUnitSelected(unit.id)) {
+            this.editMode = 'addUnit'
+            this.selectedUnit = null
+        } else {
+            this.editMode = 'editUnit'
+            this.selectedUnit = unit
+        }        
+    }
+    public getSelectedUnit (): BuildingUnit {
+        return this.selectedUnit ? this.getUnitInfo(this.selectedUnit.id) : null
+    } 
+    public isUnitSelected (id: string): boolean {
+        return this.selectedUnit && this.selectedUnit.id === id
+    }
+
+    public mapAddSection (section: BuildingSection): void {
+        let unitNumber = this.nextUnitNumber
+        const { name, minFloor, maxFloor, unitsOnFloor } = section
+        const newSection = {
+            id: String(++this.autoincrement),
+            floors: [],
+            name,
+            index: this.sections.length + 1,
+            type: 'section',
+        }
+        for (let floor = minFloor; floor <= maxFloor; floor++) {
+            if (floor === 0) {
+                continue
+            }
+            const units = []
+            for (let unitOnFloor = 0; unitOnFloor < unitsOnFloor; unitOnFloor++) {
+                let label = ' '
+                if (floor > 0) {
+                    label = String(unitNumber)
+                    unitNumber++
+                }
+                units.push({
+                    id: String(++this.autoincrement),
+                    label,
+                    type: 'unit',
+                })
+            }
+            newSection.floors.unshift({
+                id: String(++this.autoincrement),
+                index: floor,
+                name: String(floor),
+                type: 'floor',
+                units,
+            })
+        }
+        this.map.sections.push(newSection)
+        this.visibleSections.push(newSection.id)
+        this.updateFormField()
+    }
+    
+    public mapUpdateSection (section: BuildingSection): void {
+        const sectionIndex = this.map.sections.findIndex(mapSection => section.id === mapSection.id)
+        if (sectionIndex !== -1) {
+            this.map.sections[sectionIndex].name = section.name
+        }
+        this.editMode = 'addSection'
+        this.updateFormField()
+    }
+    
+    public mapRemoveSection (id: string): void {
+        const sectionIndex = this.map.sections.findIndex(mapSection => mapSection.id === id)
+        this.map.sections.splice(sectionIndex, 1)
+        this.editMode = 'addSection'
+        this.updateFormField()
+    }
+
+    public mapAddUnit (unit: BuildingUnit): void {
```

Comment:
`move verbs on the start of the method/function name's`

i'll comment this one earlier